src/Microsoft.DotNet.CodeFormatting.Tests/PropertyBagTests.cs

```diff
@@ -0,0 +1,443 @@
+            Assert.AreEqual(expectedValue, converted, "String conversion to primitive type did not succeed.");
+
+            Assert.IsFalse(propertyBag.TryGetProperty("testvalue", out converted));
+
+            Assert.IsTrue(copiedPropertyBag.TryGetProperty("TestValue", out converted));
+            Assert.AreEqual(expectedValue, converted, "String conversion to primitive type did not succeed.");
+
+            Assert.IsTrue(copiedPropertyBag.TryGetProperty("testvalue", out converted));
+            Assert.AreEqual(expectedValue, converted, "String conversion to primitive type did not succeed.");
+        }
+
+        [TestMethod]
+        public void PropertyBag_SetProperty()
+        {
+            var propertyBag = new PropertyBag();
+
+            var descriptor = new PerLanguageOption<string>("feature", "name", "defaultValue");
+
+            propertyBag.SetProperty(descriptor, "value");
+            Assert.IsTrue(propertyBag.GetProperty(descriptor) == "value");
+
+            propertyBag.SetProperty(descriptor, null);
+            Assert.IsTrue(propertyBag.GetProperty(descriptor) == "defaultValue");
+        }
+
+        [TestMethod]
+        public void PropertyBag_BinarySerialization()
+        {
+            var testData = new TestData();
+            var propertyBag = new PropertyBag();
+            testData.InitializePropertyBag(propertyBag);
+
+
+            var formatter = new BinaryFormatter();
+
+            string path = Path.GetTempFileName();
+
+            try
+            {
+                using (var writer = new FileStream(path, FileMode.OpenOrCreate, FileAccess.ReadWrite))
+                {
+                    formatter.Serialize(writer, propertyBag);
+                }
+
+                using (var reader = new FileStream(path, FileMode.OpenOrCreate, FileAccess.ReadWrite))
+                {
+                    propertyBag = (PropertyBag)formatter.Deserialize(reader);
+                    testData.ValidatePropertyBag(propertyBag);
+                }
+            }
+            finally
+            {
+                try { File.Delete(path); } catch (IOException) { }
+            }
+        }
+
+        [TestMethod]
+        public void PropertyBag_ArgumentNullChecks()        
+        {
+            bool argumentNull = false;
+
+            var propertyBag = new PropertyBag();
+            try
+            {
+                propertyBag.SetProperty((PerLanguageOption<string>)null, "string");
+            }
+            catch (ArgumentNullException) { argumentNull = true; }
+            Assert.IsTrue(argumentNull);
+
+            argumentNull = false;
+            try
+            {
+                propertyBag.GetProperty((PerLanguageOption<string>)null);
+            }
+            catch (ArgumentNullException) { argumentNull = true; }
+            Assert.IsTrue(argumentNull);
+        }
+
+        [TestMethod]
+        public void PropertyBag_HumanReadableSerialization()
```

Comment:
serialization to/and from XML == human-readable. 
serialization using binary formatter == non-readable persistence representation.

I changed name to XmlSerialization